### PR TITLE
Maintain column order when adding new columns to table

### PR DESCRIPTION
**Pull request by cythrawll**

### CODE REVIEW

**Comment by cythrawll** [comment]
any suggestions for that? the existing order tests wouldn't work for this case. wondering if it's really worth adding the testing infrastructure needed to test this thing.

**Comment by balajimanoharan** [comment]
We have a test in visualizations that does something similar.
https://github.com/yahoo/navi/blob/master/packages/visualizations/tests/acceptance/table-test.js#L23

**Comment by balajimanoharan** [comment]
https://github.com/jgwhite/ember-sortable/blob/1.x/test-support/helpers/ember-sortable/test-helpers.js#L2

**Comment by cythrawll** [comment]
tests added and found another bug due to metric change of shape

File: packages/visualizations/addon/models/table.js [hunk 1]
```diff
@@ -92,12 +92,25 @@ export default VisualizationBase.extend(Validations, {
...
+    ];
+
+    if(columns.length) {
+      newColumns.sort((a,b) => {
+        let foundA = columns.findIndex(item => isSameColumn(item, a)),
+            foundB = columns.findIndex(item => isSameColumn(item, b));
+        if(foundA === -1 || foundB === -1) {
+          return 0;
+        }
+        return foundA - foundB;
```
Comment (mattgraham1995) [inline packages/visualizations/addon/models/table.js]: Could we do this?
`return foundA > -1 && foundB > -1 ? foundA - foundB : 0;`

File: packages/visualizations/addon/models/table.js [hunk 2]
```diff
@@ -92,12 +92,25 @@ export default VisualizationBase.extend(Validations, {
       };
     });
 
+    let newColumns = [
```
Comment (balajimanoharan) [inline packages/visualizations/addon/models/table.js]: I feel `rebuildConfig` here has grown to be a bit too big. Can we split this into multiple smaller methods?
`getMetricColumns`, `getDimensionColumns`...

File: packages/visualizations/addon/models/table.js [hunk 3]
```diff
@@ -124,6 +137,21 @@ function hasAllColumns(request, columns) {
...
+ * @function isSameColumn
+ * @param {Object} colA - column object
+ * @param {Object} colB - column object
+ * @returns {Boolean} - whether the columns are the same field
+ */
+function isSameColumn(colA, colB) {
+  if(colA.type === colB.type) {
+    return colA.type === 'dimension' ? 
+      colA.field.dimension == colB.field.dimension : 
+      canonicalizeMetric(colA.field) == canonicalizeMetric(colB.field);
```
Comment (balajimanoharan) [inline packages/visualizations/addon/models/table.js]: Also I feel like this logic is done in multiple places. Could we make use of displayName here instead? `colA.displayName === colB.displayName`

**Comment by cythrawll** [comment]
refactored rebuild component, PR comments addressed

**Pull request merged**